integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOCILoadBalancer.java

```diff
@@ -0,0 +1,321 @@
+   * Check that application is accessabale via OCI LoadBalancer
+   */
+  @Test
+  @DisplayName("Test the sample-app app can be accessed"
+      + " from all managed servers in the domain through OCI Load Balancer.")
+  public void testOCILB() throws Exception {
+
+    // create and verify one cluster mii domain
+    logger.info("Create domain and verify that it's running");
+    String miiImage1 = createAndVerifyMiiImage(MODEL_DIR + "/model-singleclusterdomain-sampleapp-wls.yaml");
+    // create docker registry secret to pull the image from registry
+    // this secret is used only for non-kind cluster
+    logger.info("Create docker registry secret in namespace {0}", domain1Namespace);
+    createOcirRepoSecret(domain1Namespace);
+    createAndVerifyDomain(miiImage1, domain1Namespace, domain1Uid);
+    int clusterHttpPort = 8001;
+
+    assertDoesNotThrow(() -> installAndVerifyOCILoadBalancer(domain1Namespace,
+        clusterHttpPort, cluster1Name, domain1Uid, OCI_LB_NAME),
+        "Installation of OCI Load Balancer failed");
+    loadBalancerIP = getLoadBalancerIP(domain1Namespace,OCI_LB_NAME);
+    assertNotNull(loadBalancerIP, " External IP for Load Balancer is undefined");
+    logger.info(" LoadBalancer IP is " + loadBalancerIP);
+    verifyWebAppAccessThroughOCILB(loadBalancerIP, 2, clusterHttpPort);
+  }
+
+  /** Retreive external IP from OCI LoadBalancer.
+   *
+   */
+  private static String getLoadBalancerIP(String namespace, String lbName) throws Exception {
+    Map<String, String> labels = new HashMap<>();
+    labels.put("loadbalancer", lbName);
+    V1Service service = getService(lbName, labels, namespace);
+    assertNotNull(service, "Can't find service with name " + lbName);
+    logger.info("Found service with name {0} in {1} namespace ", lbName, namespace);
+    List<V1LoadBalancerIngress> ingress = service.getStatus().getLoadBalancer().getIngress();
+    if (ingress != null) {
+      logger.info("LoadBalancer Ingress " + ingress.toString());
+      V1LoadBalancerIngress lbIng = ingress.stream().filter(c ->
+          !c.getIp().equals("pending")
+      ).findAny().orElse(null);
+      if (lbIng != null) {
+        logger.info("OCI LoadBalancer is created with external ip" + lbIng.getIp());
+        return lbIng.getIp();
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Create mii image with sample-app application.
+   */
+  private static String createAndVerifyMiiImage(String modelFile) {
+    // create image with model files
+    logger.info("Create image with model file and verify");
+
+    List<String> appList = new ArrayList();
+    appList.add(SAMPLE_APP_NAME);
+
+    // build the model file list
+    final List<String> modelList = Collections.singletonList(modelFile);
+    String myImage =
+        createMiiImageAndVerify(IMAGE_NAME, modelList, appList);
+
+    // docker login and push image to docker registry if necessary
+    dockerLoginAndPushImageToRegistry(myImage);
+
+    return myImage;
+  }
+
+  /**
```

Comment:
Can the OCI load balancer proxy entire weblogic cluster unlike Traefik or Voyager.  Why  OCI LoadBalancer  is used to proxy individual managed server

Reply:
it is per not individual server, it is per wls cluster

Reply:
https://build.weblogick8s.org:8443/job/wko-oke-nightly-seqential/239 newest run after corrections